src/simplewallet/simplewallet.cpp

```diff
@@ -7325,7 +7325,7 @@ bool simple_wallet::wallet_info(const std::vector<std::string> &args)
 //----------------------------------------------------------------------------------------------------
 bool simple_wallet::sign(const std::vector<std::string> &args)
 {
-  if (m_wallet->key_on_device())
+  if (m_wallet->key_on_device() != hw::device::device_type::SOFTWARE)
```

Comment:
key_on_device would likely return true/false, so "!= hw::device::device_type::SOFTWARE" seems redundant, unless it's meant as a "get_key_backing_type". It'd avoid that couple dozen boilerplate changes too.

Reply:
it's meant as a "get_key_backing_type" - maybe renaming it to this would be better. problem is, it's already called key_on_device in the (existing) json keys file.

Reply:
You could add a new  one too, and seed it with the valiue of key_on_device if present. If it's a hassle then I suppose a function rename seems best. It's not like it has to be the same name as a json field.